packages/terra-navigation-side-menu/src/NavigationSideMenu.jsx

```diff
@@ -172,11 +189,11 @@ class NavigationSideMenu extends React.Component {
     );
   }
 
-  buildListContent(currentItem) {
-    if (currentItem && currentItem.childKeys && currentItem.childKeys.length) {
-      return <List className={cx(['menu-list'])}>{currentItem.childKeys.map(key => this.buildListItem(key))}</List>;
-    }
-    return null;
+  buildListContent(keys) {
+    return (keys && keys.length
+      ? <List className={cx(['menu-list'])}>{keys.map(key => this.buildListItem(key))}</List>
+      : <FormattedMessage id="Terra.NavigationSideMenu.noMatch" />
```

Comment:
Open to show something other than "No match..." here for the empty list state.

Some other ideas:
* Show nothing (seems bad)
* Show the entire, unfiltered list